third_party/terraform/resources/resource_google_project.go

```diff
@@ -173,6 +177,26 @@ func resourceGoogleProjectCreate(d *schema.ResourceData, meta interface{}) error
 	return nil
 }
 
+func resourceGoogleProjectCheckPreRequisites(config *Config, d *schema.ResourceData) error {
+	ib, ok := d.GetOk("billing_account")
+	if !ok {
+		return nil
+	}
+	ba := "billingAccounts/" + ib.(string)
+	const perm = "billing.resourceAssociations.create"
+	req := &cloudbilling.TestIamPermissionsRequest{
+		Permissions: []string{perm},
+	}
+	resp, err := config.clientBilling.BillingAccounts.TestIamPermissions(ba, req).Do()
+	if err != nil {
```

Comment:
Was there a consensus on whether this request failing should actually fail the apply?

Reply:
I changed my mind and decided it's fine. I'd love to make this a check for the correct API being enabled but I'm not sure it's possible at this point. I added something to that effect in https://github.com/terraform-providers/terraform-provider-google/pull/5671#discussion_r380980787

Reply:
Got it, thanks!